tools/relay/README.md

```diff
@@ -0,0 +1,47 @@
+# Relay
+
+Relay will act as an interface between invoker and the benchmark server, allowing the user of a function to supply a custom form of input than a strict Helloworld format (see `aes.proto` for example).
+
+## Adding relay
+
+For docker compose files, add the following snippet
+
+```yaml
+relay:
+  image: vhiveease/relay:latest
+  entrypoint:
+    - /app/server
+    - --addr=0.0.0.0:50000
+    - --function-endpoint-url=${FUNCTION_URL}
+    - --function-endpoint-port=${FUNCTION_PORT}
+    - --function-name=${FUNCTION_NAME}
+  ports:
+    - published: 50000
+      target: 50000
+```
+where FUNCTION_NAME is replaced by the name you have added to the `getclient.go` in the [vSwarm-proto repository](https://github.com/ease-lab/vSwarm-proto/blob/main/grpcclient/getclient.go). You can remove the published port of your benchmark server now (which is now FUNCTION_PORT). Generally speaking, FUNCTION_URL in same docker-compose file will be given by the container name (see aes-go file for reference).
```

Comment:
I dont really get the point with the URL. Can you make it a bit more clear?

Reply:
So when the 2 containers are running in a docker runtime, the relay would need to access the aes-go server through a URL. For that, docker assigns the URL as the container name (So since aes-go container name is aes-go, that will be the endpoint for it).

Reply:
Yes, that is how relay runs.

Reply:
I can't think of any other way to make it better. I think if someone looks at the `aes-go` function yaml files, they would understand what FUNCTION_URL is supposed to be.

Reply:
I'd wait for David to accept this change. Note that a doc should not require looking at any source code files, it should be self-contained.

Reply:
Ok so I was a bit confused about the URL description.
I would say make a reference to how docker networks work:
https://docs.docker.com/compose/networking/
You can refrase or use their first sentence maybe something like this:
_To understand the URL setting in detail check see [here](https://docs.docker.com/compose/networking/) to find out how docker network works.  
Generally speaking in this case compose sets up a single network between the relay and the function container. Now the function container is reachable from the relay via the service/container name. See [docker networking](https://docs.docker.com/compose/networking/) or aes-go as an example_